test/integration/test_dataplane.go

```diff
@@ -227,6 +294,29 @@ func TestDataPlaneEssentials(t *testing.T) {
 
 		return true
 	}, waitTime, tickTime)
+
+	t.Log("updating dataplane spec with proxy service type of NodePort")
+	require.Eventually(t,
+		testutils.DataPlaneUpdateEventually(t, GetCtx(), dataplaneName, clients, func(dp *operatorv1beta1.DataPlane) {
+			dp.Spec.Network.Services.Ingress.Type = corev1.ServiceTypeNodePort
+		}),
+		waitTime, tickTime)
+
+	t.Log("checking if dataplane proxy service type changes to NodePort")
+	require.Eventually(t, func() bool {
+		servicesClient := GetClients().K8sClient.CoreV1().Services(dataplane.Namespace)
+		dataplaneIngressService, err := servicesClient.Get(GetCtx(), dataplaneIngressService.Name, metav1.GetOptions{})
+		if err != nil {
+			t.Logf("error getting dataplane proxy service: %v", err)
+			return false
+		}
+		if dataplaneIngressService.Spec.Type != corev1.ServiceTypeNodePort {
+			t.Logf("dataplane proxy service should be of NodePort type but is %s", dataplaneIngressService.Spec.Type)
+			return false
+		}
+
+		return true
+	}, waitTime, tickTime)
```

Comment:
I believe that this block together with the one above can be factored out and put in a `func` which would be called twice (ideally in a table driven test) with `corev1.ServiceTypeClusterIP` and `corev1.ServiceTypeNodePort`. WDYT?